src/main/java/org/nguh/nguhcraft/block/GrapeCropBlock.kt

```diff
@@ -0,0 +1,147 @@
+package org.nguh.nguhcraft.block
+
+import net.minecraft.core.BlockPos
+import net.minecraft.sounds.SoundEvents
+import net.minecraft.sounds.SoundSource
+import net.minecraft.world.InteractionHand
+import net.minecraft.world.InteractionResult
+import net.minecraft.world.entity.player.Player
+import net.minecraft.world.item.ItemStack
+import net.minecraft.world.item.Items
+import net.minecraft.world.item.context.BlockPlaceContext
+import net.minecraft.world.level.*
+import net.minecraft.world.level.block.Block
+import net.minecraft.world.level.block.CropBlock
+import net.minecraft.world.level.block.state.BlockState
+import net.minecraft.world.level.block.state.StateDefinition
+import net.minecraft.world.level.block.state.properties.BlockStateProperties
+import net.minecraft.world.level.block.state.properties.BooleanProperty
+import net.minecraft.world.level.block.state.properties.IntegerProperty
+import net.minecraft.world.level.gameevent.GameEvent
+import net.minecraft.world.phys.BlockHitResult
+import net.minecraft.world.phys.shapes.CollisionContext
+import net.minecraft.world.phys.shapes.VoxelShape
+import org.nguh.nguhcraft.item.NguhItems
+
+class GrapeCropBlock(settings: Properties) : CropBlock(settings) {
+    fun init() {
+        this.registerDefaultState(super.defaultBlockState().setValue(this.getStickLoggedProperty(), true))
+    }
+
+    override fun getBaseSeedId(): ItemLike {
+        return NguhItems.GRAPE_SEEDS
+    }
+
+    override fun getAgeProperty(): IntegerProperty {
+        return AGE
+    }
+
+    override fun getMaxAge(): Int {
+        return MAX_AGE
+    }
+
+    fun getStickLoggedProperty(): BooleanProperty {
+        return STICK_LOGGED
+    }
```

Comment:
I don’t think this function needs to exist considering that you can just access `STICK_LOGGED` directly.